### DISCUSSION TITLE
Rave-4-Win64

### DISCUSSION THREAD

**Discussion by cumsoft** [acids-ircam/RAVE#210] [General]
Hey All,

Figured *We* and *Us* would take a little time to play around with porting/migration RAVE to Windows.

We are sure as most of you can guess the Python framework is still very new to the Windows.  Just a quick google would inform that Python isn't backwards compatible with windows 7 and earlier and still very much in development for Windows 8, 10, 11 etc..

Our first speed bump, 
Python is still rolling therefore the most uptodate packages for the framework might not be fully accessible or compatible with secondary operating systems like Windows. In our findings python 3.10.0 & 3.11.0 seem to be most stable. Although It is mentioned that 3.12 is on its way. So really in closing alot of the issues faced are really package compatibility ie. which version of python in conjunction with which pip version will effect Conda environment when porting to Windows. 

*Lil note, remember that windows is just C with c# and of course other langs so boiled down its really simply how do we launch RAVE (Python) from C.
_____________________________________________________________________________________________________

Another different tangent is concerns with runtime errors. We are positive that many of ya'll are most likely much more skilled than us in python; which is why its best to take these concerns with a grain of salt. We didn't really take much time to look at the source in linux, what we did do is tried implementing the RAVE framework in Windows via automized .ps1.  Duh.

Okay so our hunch is basically two fold, the first is that when contemplating the RAVE framework stack one might imagine installation being something like (Install python --> install pip ---> install conda --> install RAVE) or mabye (Install python --> install pip --> install RAVE). We don't know if this is true and the exact stack implementing however you might find that its actually alot different/complex. Weirdly enough you can install python from pip. So our advice is to be cognizant of this. Meaning perhaps the stack should be something along the lines of (install python --> install Conda  (python virtenv) --> install pip --> install RAVE).  This might resolve runtime issues perhaps.

The second is similar when dealing with the stack; in that how your python library is managed. Hopefully properly lol. This becomes an issue when new framework versions upgrade or change. Meaning simply, 

superusers python packages should be in a folder like C:\Python27\Scripts when a new framework is upgraded the python package folder may now reflect a different naming convention. something like this... C:\Python38-20\Scripts. So just be mindful of this. Definitely a maintainers job haha. So like, you might be upgraded to the latest python version but really the RAVE syntax might just be pointing at the older folders. So setting directory pointers is important.
_____________________________________________________________________________________________________

We'd love to hear your thoughts? and again, our time spent with this framework is limited so please take this with a grain of salt.

Below attached is an attempt at automating RAVE package for windows with a simple powershell script. which ultimately could be implimented in the RAVE framework package. 

 Github doesn't allow us to upload a .ps1 *LOL* so heres a plain-text.

Enjoy the rest of the week plz 
With <3
[RAVE4win_CumsoftTweak.txt](https://github.com/acids-ircam/RAVE/files/10865442/RAVE4win_CumsoftTweak.txt)

**Comment by ogbabydiesal** [comment]
That was really helpful for this conda-venv noob thanks. Ended up sorting out my issue with this guidance. Kept stalling at pretraining, fixed by deleting all versions of Python on my windows laptop, installing conda, creating a virtual environment with ffmpeg, launching pycharm, firing up the terminal and installing acids-rave after confirming my pycharm was targeting my new environment (also be sure that your terminal is in that environment). Preprocessing now works, now I just need to get my PyTorch to recognize my nvidea 😩
